src/commit_log.rs

```diff
@@ -119,8 +122,76 @@ pub trait CommitLog {
     fn scoped_subscribe<'a>(
         &'a self,
         consumer_group_name: &str,
-        offsets: Option<Vec<ConsumerOffset>>,
+        offsets: Vec<ConsumerOffset>,
         subscriptions: Vec<Subscription>,
         idle_timeout: Option<Duration>,
     ) -> Pin<Box<dyn Stream<Item = ConsumerRecord> + Send + 'a>>;
 }
+
+fn nullable_vec<'de, D, T>(d: D) -> Result<Vec<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de>,
+{
+    Deserialize::deserialize(d).map(|x: Option<_>| x.unwrap_or_default())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_nullable_vec_handles_null() {
+        let json = r#"
+        {
+            "offsets": null,
```

Comment:
What about another case where the field is not null, but completely absent?

Reply:
Feels like a test of serde itself?

Reply:
Not sure. Might be necessary to add some serde attibutes to make it work?